### PR TITLE
Keep HTTP connections to the Icinga 2 API alive

### LINKED ISSUE #3001: Keep HTTP connections to the Icinga 2 API alive
This would save us a lot of expensive TLS handshakes.

### PR DESCRIPTION
**Pull request by Al2Klimov**
fixes #3001

### CODE REVIEW

File: modules/monitoring/library/Monitoring/Web/Rest/RestRequest.php [hunk 1]
```diff
@@ -209,6 +216,7 @@ public function send()
             "Host: {$url['host']}",
             "Content-Type: {$this->contentType}",
             'Accept: application/json',
+            'Connection: keep-alive',
```
Comment (dnsmichi) [inline modules/monitoring/library/Monitoring/Web/Rest/RestRequest.php]: Imho this shouldn't be hardcoded in this generic class, but left to the caller (e.g. the command transport).

Comment (Al2Klimov) [reply modules/monitoring/library/Monitoring/Web/Rest/RestRequest.php]: @dnsmichi Some other headers *are* already hardcoded – so what.

Comment (dnsmichi) [reply modules/monitoring/library/Monitoring/Web/Rest/RestRequest.php]: It is a behavioural change. I wouldn't go with that argument of other default headers also being hardcoded but rather make it optional.

**Comment by lippserd** [comment]
We should wait for the release of `ipl-http` and use this lib instead of the `RestRequest`.

**Pull request closed without merging**